packages/prepo-ui/src/components/CurrencyInput/CurrencyInput.tsx

```diff
@@ -196,13 +197,19 @@ const CurrencyInput: React.FC<
       </Flex>
       {showBalance && (
         <Balance alignSelf="flex-end" height={16}>
-          {balance !== undefined && !balanceNotVisible && (
+          {!balanceNotVisible && (
             <>
-              {`Balance: ${displayDecimals(balance)}`}
-              {value !== balanceAfterGas && !isBalanceZero && (
-                <MaxButton disabled={disabled} onClick={handleMax}>
-                  MAX
-                </MaxButton>
+              {balance === undefined ? (
+                <Skeleton width={140} height={16} />
+              ) : (
```

Comment:
Could we have a loading prop instead, which is only true if connected && balance === undefined

currently it shows loading for non connected account
```suggestion
              {loading ? (
                <Skeleton width={140} height={16} />
              ) : (
```